lib/exercises.rb

```diff
@@ -2,29 +2,179 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) where n is the number of strings
+# Space Complexity: O(n) where n is the number of strings
+def grouped_anagrams_version_1(strings)
```

Comment:
This is a really good heuristic, but I'm uncertain if the keys you're generating are always going to be unique.  The BigO is correct, if the strings are guaranteed to be within a certain length.